folium/map.py

```diff
@@ -421,12 +418,13 @@ def _get_self_bounds(self) -> TypeBoundsReturn:
         return cast(TypeBoundsReturn, [self.location, self.location])
 
     def render(self):
-        if self.icon:
-            self.add_child(self.SetIcon(marker=self, icon=self.icon))
         if self.location is None:
             raise ValueError(
                 f"{self._name} location must be assigned when added directly to map."
             )
+        for child in list(self._children.values()):
```

Comment:
This looks strange to me. A marker can have only one `Icon`, correct?  Why do we need to look through all the children? Would it not be easier to use `self.icon`?

Reply:
> Would it not be easier to use self.icon?

That would not work for the case where the icon is added using `marker.add_child(icon)` or `icon.add_to(marker)`.